pkg/bpf/map_linux.go

```diff
@@ -550,17 +556,17 @@ type MapValidator func(path string) (bool, error)
 // deepcopy of the key and value on between each iterations to avoid memory
 // corruption.
 func (m *Map) DumpWithCallback(cb DumpCallback) error {
+	if err := m.Open(); err != nil {
```

Comment:
Since `Open` will lock/unlock `m.lock` as well, it might make sense to use `m.open` instead and move it below the lock/defer unlock below (line 563/564)?

Reply:
It uses `RLock()` so it would not protect write access to `m.fd.`

Reply:
Nevermind, sorry.